### ISSUE TITLE
InvocationTargetException when installing missing dependencies in OPR ("big button")

### ISSUE THREAD

**Issue by qtran-n4** [eclipse/n4js#956]
Compiling n4js-libs using HLC works fine. However, when importing n4js-libs into the IDE, exceptions are thrown.

**Comment by lbeurerkellner** [comment]
While one of the known causes for this issue was fixed in #967, another cause we could observe a few times so far, is indicated by the following stacktrace: 

```

java.lang.reflect.InvocationTargetException
	at org.eclipse.jface.operation.ModalContext.run(ModalContext.java:398)
	at org.eclipse.jface.wizard.WizardDialog.run(WizardDialog.java:980)
	at org.eclipse.n4js.ui.wizard.dependencies.ExternalLibrariesWizard.performFinish(ExternalLibrariesWizard.java:72)
	at org.eclipse.jface.wizard.WizardDialog.finishPressed(WizardDialog.java:778)
	at org.eclipse.jface.wizard.WizardDialog.buttonPressed(WizardDialog.java:417)
	at org.eclipse.jface.dialogs.Dialog.lambda$0(Dialog.java:619)
	at org.eclipse.swt.events.SelectionListener$1.widgetSelected(SelectionListener.java:81)
	at org.eclipse.swt.widgets.TypedListener.handleEvent(TypedListener.java:249)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:86)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4258)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1502)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1525)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1510)
	at org.eclipse.swt.widgets.Widget.notifyListeners(Widget.java:1314)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4082)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3699)
	at org.eclipse.jface.window.Window.runEventLoop(Window.java:818)
	at org.eclipse.jface.window.Window.open(Window.java:794)
	at org.eclipse.n4js.ui.handler.LibrariesFixHandler.execute(LibrariesFixHandler.java:26)
	at org.eclipse.ui.internal.handlers.HandlerProxy.execute(HandlerProxy.java:291)
	at org.eclipse.ui.internal.handlers.E4HandlerProxy.execute(E4HandlerProxy.java:92)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.e4.core.internal.di.MethodRequestor.execute(MethodRequestor.java:55)
	at org.eclipse.e4.core.internal.di.InjectorImpl.invokeUsingClass(InjectorImpl.java:305)
	at org.eclipse.e4.core.internal.di.InjectorImpl.invoke(InjectorImpl.java:239)
	at org.eclipse.e4.core.contexts.ContextInjectionFactory.invoke(ContextInjectionFactory.java:132)
	at org.eclipse.e4.core.commands.internal.HandlerServiceHandler.execute(HandlerServiceHandler.java:152)
	at org.eclipse.core.commands.Command.executeWithChecks(Command.java:494)
	at org.eclipse.core.commands.ParameterizedCommand.executeWithChecks(ParameterizedCommand.java:487)
	at org.eclipse.e4.core.commands.internal.HandlerServiceImpl.executeHandler(HandlerServiceImpl.java:210)
	at org.eclipse.e4.ui.workbench.renderers.swt.HandledContributionItem.executeItem(HandledContributionItem.java:431)
	at org.eclipse.e4.ui.workbench.renderers.swt.AbstractContributionItem.handleWidgetSelection(AbstractContributionItem.java:446)
	at org.eclipse.e4.ui.workbench.renderers.swt.AbstractContributionItem.lambda$2(AbstractContributionItem.java:472)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:86)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4258)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1502)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1525)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1510)
	at org.eclipse.swt.widgets.Widget.notifyListeners(Widget.java:1314)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4082)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3699)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1150)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1039)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:153)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:680)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:594)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:148)
	at org.eclipse.n4js.product.N4JSApplication.start(N4JSApplication.java:141)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:388)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:243)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:653)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:590)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1499)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1472)
Caused by: org.eclipse.xtext.builder.resourceloader.IResourceLoader$LoadOperationException: java.lang.IllegalStateException: This stopwatch is already stopped.
	at org.eclipse.xtext.builder.resourceloader.ParallelResourceLoader$ParallelLoadOperation.next(ParallelResourceLoader.java:179)
	at org.eclipse.xtext.builder.resourceloader.AbstractResourceLoader$CheckedLoadOperation.next(AbstractResourceLoader.java:77)
	at org.eclipse.n4js.ui.building.BuildScopeAwareParallelLoader$1.next(BuildScopeAwareParallelLoader.java:121)
	at org.eclipse.n4js.ui.building.WriteNewResourceDescriptionsImplementation.doWriteResourceDescriptions(WriteNewResourceDescriptionsImplementation.java:105)
	at org.eclipse.n4js.ui.building.WriteNewResourceDescriptionsImplementation.writeNewResourceDescriptions(WriteNewResourceDescriptionsImplementation.java:84)
	at org.eclipse.n4js.ui.building.N4ClusteringBuilderState.writeNewResourceDescriptions(N4ClusteringBuilderState.java:146)
	at org.eclipse.n4js.ui.building.DoUpdateImplementation.writeNewResourceDescriptions(DoUpdateImplementation.java:388)
	at org.eclipse.n4js.ui.building.DoUpdateImplementation.doUpdate(DoUpdateImplementation.java:110)
	at org.eclipse.n4js.ui.building.N4ClusteringBuilderState.doUpdate(N4ClusteringBuilderState.java:105)
	at org.eclipse.n4js.ui.building.N4JSGenerateImmediatelyBuilderState.doUpdate(N4JSGenerateImmediatelyBuilderState.java:203)
	at org.eclipse.xtext.builder.builderState.AbstractBuilderState.update(AbstractBuilderState.java:116)
	at org.eclipse.n4js.ui.external.ExternalLibraryBuilder$BuildOperation.run(ExternalLibraryBuilder.java:471)
	at org.eclipse.n4js.ui.external.ExternalLibraryBuilder$BuildOperation.access$3(ExternalLibraryBuilder.java:429)
	at org.eclipse.n4js.ui.external.ExternalLibraryBuilder.doPerformOperation(ExternalLibraryBuilder.java:301)
	at org.eclipse.n4js.ui.external.ExternalLibraryBuilder.build(ExternalLibraryBuilder.java:178)
	at org.eclipse.n4js.ui.external.ExternalLibraryBuilder.build(ExternalLibraryBuilder.java:160)
	at org.eclipse.n4js.ui.external.EclipseExternalLibraryWorkspace.registerProjectsInternal(EclipseExternalLibraryWorkspace.java:320)
	at org.eclipse.n4js.ui.external.EclipseExternalLibraryWorkspace.registerProjects(EclipseExternalLibraryWorkspace.java:211)
	at org.eclipse.n4js.ui.external.EclipseExternalIndexSynchronizer.synchronizeIndex(EclipseExternalIndexSynchronizer.java:139)
	at org.eclipse.n4js.ui.external.EclipseExternalIndexSynchronizer.synchronizeNpms(EclipseExternalIndexSynchronizer.java:90)
	at org.eclipse.n4js.external.LibraryManager.installNPMsInternal(LibraryManager.java:235)
	at org.eclipse.n4js.external.LibraryManager.lambda$2(LibraryManager.java:197)
	at org.eclipse.n4js.external.LibraryManager.runWithWorkspaceLock(LibraryManager.java:595)
	at org.eclipse.n4js.external.LibraryManager.installNPMs(LibraryManager.java:197)
	at org.eclipse.n4js.ui.external.ExternalLibrariesActionsHelper.installNoUpdate(ExternalLibrariesActionsHelper.java:132)
	at org.eclipse.n4js.ui.wizard.dependencies.ExternalLibrariesInstallHelper.calculateAndInstallDependencies(ExternalLibrariesInstallHelper.java:67)
	at org.eclipse.n4js.ui.wizard.dependencies.RunnableInstallDependencies.run(RunnableInstallDependencies.java:88)
	at org.eclipse.jface.operation.ModalContext$ModalContextThread.run(ModalContext.java:119)
Caused by: java.lang.IllegalStateException: This stopwatch is already stopped.
	at com.google.common.base.Preconditions.checkState(Preconditions.java:173)
	at com.google.common.base.Stopwatch.stop(Stopwatch.java:175)
	at org.eclipse.n4js.smith.TimedMeasurement.end(TimedMeasurement.java:38)
	at org.eclipse.n4js.smith.TimedMeasurement.close(TimedMeasurement.java:46)
	at org.eclipse.n4js.resource.N4JSLinker$1.process(N4JSLinker.java:119)
	at org.eclipse.n4js.resource.N4JSLinker$1.process(N4JSLinker.java:1)
	at org.eclipse.xtext.util.concurrent.IUnitOfWork$Void.exec(IUnitOfWork.java:37)
	at org.eclipse.xtext.util.OnChangeEvictingCache.execWithoutCacheClear(OnChangeEvictingCache.java:129)
	at org.eclipse.n4js.resource.N4JSLinker.doLinkModel(N4JSLinker.java:101)
	at org.eclipse.xtext.linking.impl.AbstractCleaningLinker.linkModel(AbstractCleaningLinker.java:51)
	at org.eclipse.xtext.resource.XtextResource.doLinking(XtextResource.java:340)
	at org.eclipse.xtext.linking.lazy.LazyLinkingResource.doLinking(LazyLinkingResource.java:107)
	at org.eclipse.n4js.resource.N4JSResource.updateInternalState(N4JSResource.java:599)
	at org.eclipse.xtext.resource.XtextResource.updateInternalState(XtextResource.java:290)
	at org.eclipse.xtext.resource.DerivedStateAwareResource.updateInternalState(DerivedStateAwareResource.java:161)
	at org.eclipse.n4js.resource.N4JSResource.doLoad(N4JSResource.java:629)
	at org.eclipse.emf.ecore.resource.impl.ResourceImpl.load(ResourceImpl.java:1518)
	at org.eclipse.emf.ecore.resource.impl.ResourceImpl.load(ResourceImpl.java:1297)
	at org.eclipse.xtext.resource.persistence.StorageAwareResource.load(StorageAwareResource.java:80)
	at org.eclipse.n4js.resource.N4JSResource.superLoad(N4JSResource.java:535)
	at org.eclipse.n4js.resource.N4JSResource.load(N4JSResource.java:809)
	at org.eclipse.emf.ecore.resource.impl.ResourceSetImpl.demandLoad(ResourceSetImpl.java:259)
	at org.eclipse.emf.ecore.resource.impl.ResourceSetImpl.demandLoadHelper(ResourceSetImpl.java:274)
	at org.eclipse.xtext.resource.XtextResourceSet.getResource(XtextResourceSet.java:265)
	at org.eclipse.xtext.builder.resourceloader.AbstractResourceLoader.loadResource(AbstractResourceLoader.java:44)
	at org.eclipse.xtext.builder.resourceloader.ParallelResourceLoader.loadResource(ParallelResourceLoader.java:94)
	at org.eclipse.xtext.builder.resourceloader.ParallelResourceLoader$ParallelLoadOperation$ResourceLoadJob.run(ParallelResourceLoader.java:141)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:748)
Root exception:
org.eclipse.xtext.builder.resourceloader.IResourceLoader$LoadOperationException: java.lang.IllegalStateException: This stopwatch is already stopped.
	at org.eclipse.xtext.builder.resourceloader.ParallelResourceLoader$ParallelLoadOperation.next(ParallelResourceLoader.java:179)
	at org.eclipse.xtext.builder.resourceloader.AbstractResourceLoader$CheckedLoadOperation.next(AbstractResourceLoader.java:77)
	at org.eclipse.n4js.ui.building.BuildScopeAwareParallelLoader$1.next(BuildScopeAwareParallelLoader.java:121)
	at org.eclipse.n4js.ui.building.WriteNewResourceDescriptionsImplementation.doWriteResourceDescriptions(WriteNewResourceDescriptionsImplementation.java:105)
	at org.eclipse.n4js.ui.building.WriteNewResourceDescriptionsImplementation.writeNewResourceDescriptions(WriteNewResourceDescriptionsImplementation.java:84)
	at org.eclipse.n4js.ui.building.N4ClusteringBuilderState.writeNewResourceDescriptions(N4ClusteringBuilderState.java:146)
	at org.eclipse.n4js.ui.building.DoUpdateImplementation.writeNewResourceDescriptions(DoUpdateImplementation.java:388)
	at org.eclipse.n4js.ui.building.DoUpdateImplementation.doUpdate(DoUpdateImplementation.java:110)
	at org.eclipse.n4js.ui.building.N4ClusteringBuilderState.doUpdate(N4ClusteringBuilderState.java:105)
	at org.eclipse.n4js.ui.building.N4JSGenerateImmediatelyBuilderState.doUpdate(N4JSGenerateImmediatelyBuilderState.java:203)
	at org.eclipse.xtext.builder.builderState.AbstractBuilderState.update(AbstractBuilderState.java:116)
	at org.eclipse.n4js.ui.external.ExternalLibraryBuilder$BuildOperation.run(ExternalLibraryBuilder.java:471)
	at org.eclipse.n4js.ui.external.ExternalLibraryBuilder$BuildOperation.access$3(ExternalLibraryBuilder.java:429)
	at org.eclipse.n4js.ui.external.ExternalLibraryBuilder.doPerformOperation(ExternalLibraryBuilder.java:301)
	at org.eclipse.n4js.ui.external.ExternalLibraryBuilder.build(ExternalLibraryBuilder.java:178)
	at org.eclipse.n4js.ui.external.ExternalLibraryBuilder.build(ExternalLibraryBuilder.java:160)
	at org.eclipse.n4js.ui.external.EclipseExternalLibraryWorkspace.registerProjectsInternal(EclipseExternalLibraryWorkspace.java:320)
	at org.eclipse.n4js.ui.external.EclipseExternalLibraryWorkspace.registerProjects(EclipseExternalLibraryWorkspace.java:211)
	at org.eclipse.n4js.ui.external.EclipseExternalIndexSynchronizer.synchronizeIndex(EclipseExternalIndexSynchronizer.java:139)
	at org.eclipse.n4js.ui.external.EclipseExternalIndexSynchronizer.synchronizeNpms(EclipseExternalIndexSynchronizer.java:90)
	at org.eclipse.n4js.external.LibraryManager.installNPMsInternal(LibraryManager.java:235)
	at org.eclipse.n4js.external.LibraryManager.lambda$2(LibraryManager.java:197)
	at org.eclipse.n4js.external.LibraryManager.runWithWorkspaceLock(LibraryManager.java:595)
	at org.eclipse.n4js.external.LibraryManager.installNPMs(LibraryManager.java:197)
	at org.eclipse.n4js.ui.external.ExternalLibrariesActionsHelper.installNoUpdate(ExternalLibrariesActionsHelper.java:132)
	at org.eclipse.n4js.ui.wizard.dependencies.ExternalLibrariesInstallHelper.calculateAndInstallDependencies(ExternalLibrariesInstallHelper.java:67)
	at org.eclipse.n4js.ui.wizard.dependencies.RunnableInstallDependencies.run(RunnableInstallDependencies.java:88)
	at org.eclipse.jface.operation.ModalContext$ModalContextThread.run(ModalContext.java:119)
Caused by: java.lang.IllegalStateException: This stopwatch is already stopped.
	at com.google.common.base.Preconditions.checkState(Preconditions.java:173)
	at com.google.common.base.Stopwatch.stop(Stopwatch.java:175)
	at org.eclipse.n4js.smith.TimedMeasurement.end(TimedMeasurement.java:38)
	at org.eclipse.n4js.smith.TimedMeasurement.close(TimedMeasurement.java:46)
	at org.eclipse.n4js.resource.N4JSLinker$1.process(N4JSLinker.java:119)
	at org.eclipse.n4js.resource.N4JSLinker$1.process(N4JSLinker.java:1)
	at org.eclipse.xtext.util.concurrent.IUnitOfWork$Void.exec(IUnitOfWork.java:37)
	at org.eclipse.xtext.util.OnChangeEvictingCache.execWithoutCacheClear(OnChangeEvictingCache.java:129)
	at org.eclipse.n4js.resource.N4JSLinker.doLinkModel(N4JSLinker.java:101)
	at org.eclipse.xtext.linking.impl.AbstractCleaningLinker.linkModel(AbstractCleaningLinker.java:51)
	at org.eclipse.xtext.resource.XtextResource.doLinking(XtextResource.java:340)
	at org.eclipse.xtext.linking.lazy.LazyLinkingResource.doLinking(LazyLinkingResource.java:107)
	at org.eclipse.n4js.resource.N4JSResource.updateInternalState(N4JSResource.java:599)
	at org.eclipse.xtext.resource.XtextResource.updateInternalState(XtextResource.java:290)
	at org.eclipse.xtext.resource.DerivedStateAwareResource.updateInternalState(DerivedStateAwareResource.java:161)
	at org.eclipse.n4js.resource.N4JSResource.doLoad(N4JSResource.java:629)
	at org.eclipse.emf.ecore.resource.impl.ResourceImpl.load(ResourceImpl.java:1518)
	at org.eclipse.emf.ecore.resource.impl.ResourceImpl.load(ResourceImpl.java:1297)
	at org.eclipse.xtext.resource.persistence.StorageAwareResource.load(StorageAwareResource.java:80)
	at org.eclipse.n4js.resource.N4JSResource.superLoad(N4JSResource.java:535)
	at org.eclipse.n4js.resource.N4JSResource.load(N4JSResource.java:809)
	at org.eclipse.emf.ecore.resource.impl.ResourceSetImpl.demandLoad(ResourceSetImpl.java:259)
	at org.eclipse.emf.ecore.resource.impl.ResourceSetImpl.demandLoadHelper(ResourceSetImpl.java:274)
	at org.eclipse.xtext.resource.XtextResourceSet.getResource(XtextResourceSet.java:265)
	at org.eclipse.xtext.builder.resourceloader.AbstractResourceLoader.loadResource(AbstractResourceLoader.java:44)
	at org.eclipse.xtext.builder.resourceloader.ParallelResourceLoader.loadResource(ParallelResourceLoader.java:94)
	at org.eclipse.xtext.builder.resourceloader.ParallelResourceLoader$ParallelLoadOperation$ResourceLoadJob.run(ParallelResourceLoader.java:141)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by mor-n4** [comment]
The issue reported by Luca, above, has not been observed again. The other issues have been fixed.

Closing this as FIXED.
